### ISSUE TITLE
Escaped tildes aren't treated as escaped

### ISSUE THREAD

**Issue by James-E-A** [themotte/rDrama#736]
If you type something like

```markdown
Test\~ Wow\~!
```

The site *should* render it as:

> Test\~ Wow\~!

However, it *instead* renders it as:

> Test<s> Wow</s>!

Curiously, this bug only shows up after a comment is **actually posted**; it's an observable difference between the renderer used in preview pane and the actual renderer.

**Comment by zorbathut** [comment]
Yeah, unfortunately there's a pretty big mismatch between the preview renderer and the actual renderer :/ The main tracker for that is at https://github.com/themotte/rDrama/issues/266 and I'm kinda just accumulating issues right now, this one now added to the list :)

Many thanks for the report!

**Comment by BarryOgg** [comment]
I've grappled with this yesterday, and you may be surprised to learn that converting tildes, single or double, to `<del>` is not actually part of the CommonMark standard, afaict. I think it might have been popularized by GitHub itself.

Anyways, the [mistletoe](https://github.com/miyuchina/mistletoe) renderer we're using handles double tildes, while single ones are dealt with with our own regexes, which resulted in a separate host of problems like #737. That issue now has it's band-aid solution, but this one's still broken. I propose one of two solutions:

* To hell with it, double tildes only. Old comments won't get broken as the html is parsed on save. We may add a strikethrough button next to the bold and italic ones, to not trip people up with the syntax.
* Try and extend mistletoe's renderer to handle single tildes. I think this won't re-break 737, as its tokenizer is capable of handling most other special characters in urls.

Also I really dislike how PR #764's tests don't actually import the "tested" code and contain only a copy of it, if we're having issues with imports these should be taken care of with some faker library.

**Comment by zorbathut** [comment]
I do think that a lot of this comes down to doing #266 first; we should try to find a pair of py/js parsers that can be customized to be identical, then put lots of tests in to ensure they stay that way, then we can maintain them from there (vendoring the two of them if we need to). This is obviously a Bigger Task but it would also solve the real issue.

> Also I really dislike how PR https://github.com/themotte/rDrama/pull/764's tests don't actually import the "tested" code and contain only a copy of it

Oof, I didn't notice that. Yeah that's pretty bad.

**Comment by BarryOgg** [comment]
I've tried and implemented the mistletoe extension yesterday, and I think it's mostly alright, so far e.g.
`nope ~~double~~ ~single~ ~~~triple~~~ \~escaped\~ \\~~doubleescaped\\~~`
renders as

<img width="363" height="26" alt="Image" src="https://github.com/user-attachments/assets/d37cf28d-24a8-43b5-a84a-823201842a8c" />

Which diverges from the JS version still, but hey, at least it solves _this_ issue (and links work, too).

For the record, GitHub parses that as:

nope ~~double~~ ~single~ ~~~triple~~~ \~escaped\~ \\~~doubleescaped\\~~

I'll create a PR once I update the test suite and figure out why the first \\ gets eaten.

(And eventually try and work on the long-term solution you've mentioned. Although it's worth noting that the current JS preview breaks on tildes in links, so the divergence is in our favor too sometimes.)

**Comment by zorbathut** [comment]
I would definitely say don't spend excessive effort on patches to a system that we want to nuke anyway :V but if you find weird edge cases, go add them to #266 so they can be rolled into a big test suite.
